pkgs/applications/editors/neovim/tests/default.nix

```diff
@@ -222,4 +222,27 @@ rec {
     export HOME=$TMPDIR
     ${nvimWithLuaPackages}/bin/nvim -i NONE --noplugin -es
   '';
+
+  # nixpkgs should install optional packages in the opt folder
+  nvim_with_opt_plug = neovim.override {
+    extraName = "-with-opt-plug";
+    configure.packages.plugins = with pkgs.vimPlugins; {
+      opt = [
+        (dashboard-nvim.overrideAttrs(old: { pname = old.pname + "-unique-for-tests-please-dont-use-opt"; }))
+      ];
+    };
+    configure.customRC = ''
+      try
+        Dashboard
+      catch /^Vim\%((\a\+)\)\=:E492/
+        echo "Dashboard not found"
```

Comment:
this test always succeeds doesnt it ? we should fail if Dashboard succeeds (`cquit 1` for isntance in the try segment).
And ideally we should rerun it after the packadd.

Reply:
I tried to set it so that the test failed if Dashboard autostarted, and to test it, I tried to load Dashboard in the start folder. However, Dashboard will not be found, even if it is loaded on start. Plugins are autoloaded after init.vim.

Reply:
Test fails on autoload or on failed lazyload, succeeds on successful lazyload.